sympy/solvers/diophantine/__init__.py

```diff
@@ -0,0 +1,5 @@
+from .diophantine import diophantine, classify_diop
+
+__all__ = [
+    'diophantine', 'classify_diop'
+]
```

Comment:
I don't know the diophantine module so well. Do you have any sense for whether people are likely to be using any of the other functions?

Reply:
It seems like a lot of imports have been changed in docstrings.

Looking at the docs:
https://docs.sympy.org/1.5.1/modules/solvers/diophantine.html
I see that it is suggested to use `diop_linear`

Reply:
I see that the docs do distinguish between "User functions" and "Internal functions" but the distinction is somewhat muddle because first there are many examples of using the "internal functions".

Reply:
The only place in the actual codebase that uses the module at all is in `intersection.py` in the sets package. And the documentation does explicitly say that "These functions are intended for internal use in the Diophantine module." for all the other functions.

Some of the internal functions might be able to be moved to the `ntheory` directory if they are generally useful.

In the interest of creating a clean API I would prefer to not export the equation type functions such as `diop_quadratic`, since they are called from the main `diophantine` solver without having to predetermine what the equation type is.

Reply:
Fair enough. I think that the docs should be made a bit more explicit though